onnxruntime/core/providers/webnn/builders/impl/logical_op_builder.cc

```diff
@@ -68,11 +60,25 @@ bool LogicalOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& /* initiali
   const auto& name = node.Name();
   const auto& op_type = node.OpType();
   const auto& input_defs = node.InputDefs();
-  if (input_defs.size() < 2 && op_type != "Not") {
-    LOGS(logger, VERBOSE) << op_type << " [" << name << "] requires at least 2 inputs, actual: "
-                          << input_defs.size();
-    return false;
+
+  if (input_defs.size() == 0 || input_defs.size() > 2) {
+      LOGS(logger, VERBOSE) << op_type << " [" << name << "] requires 1 or 2 inputs, actual: "
+                            << input_defs.size();
+      return false;
   }
+
+  if (input_defs.size() == 1 && op_type != "Not") {
+      LOGS(logger, VERBOSE) << op_type << " [" << name << "] requires 2 inputs, actual: "
+                            << input_defs.size();
+      return false;
+  }
+
+  if (input_defs.size() == 2 && op_type == "Not") {
+      LOGS(logger, VERBOSE) << "Not [" << name << "] requires 1 input, actual: "
+                            << input_defs.size();
+      return false;
+  }
```

Comment:
I feel this would be cleaner/shorter to read if rearranged some. e.g.

```suggestion
  size_t expected_input_count = (op_type == "Not") ? 1 : 2;
  if (input_defs.size() != expected_input_count) {
    LOGS(logger, VERBOSE) << op_type << " [" << name << "] expected input count: "
                          << expected_input_count << ", actual: " << input_defs.size();
    return false;
  }
```

Reply:
Good idea! Thanks!